press/api/client.py

```diff
@@ -115,6 +117,37 @@ def get_list(
 				.where(ParentDocType.team == frappe.local.team().name)
 			)
 
+	if doctype in ["Site", "Release Group", "Server"] and not frappe.local.system_user():
+
+		# check if user has a role and the role has any allowed items
+		if roles := get_permission_roles():
+			if frappe.get_all(
+				"Press Role Permission",
+				fields=["name"],
+				filters={"role": ("in", roles)},
+			):
```

Comment:
two queries just to check if filter should be applied or not?

Reply:
it was earlier a single check but it wasn't very readable hence I changed it to this. ref: https://github.com/frappe/press/pull/1770/commits/1d08a8e4e62a699eefc9838f12741db69d9dec0d